Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java

```diff
@@ -559,17 +568,25 @@ protected Sheet createSheet() {
             sheet.put(sheetSet);
         }
 
-        /*
-         * Add the name of the source content of the artifact represented by
-         * this node to the sheet. The value of this property is the same as the
-         * display name of the node and this a "special" property that displays
-         * the node's icon as well as the display name.
-         */
-        sheetSet.put(new NodeProperty<>(
-                Bundle.BlackboardArtifactNode_createSheet_srcFile_name(),
-                Bundle.BlackboardArtifactNode_createSheet_srcFile_displayName(),
-                NO_DESCR,
-                getDisplayName()));
+        boolean scoHasBeenAdded = false;
+        if (BlackboardArtifact.Category.ANALYSIS_RESULT == artifactType.getCategory()
+                && !(artifactType.getTypeID() == BlackboardArtifact.ARTIFACT_TYPE.TSK_HASHSET_HIT.getTypeID()
+                || artifactType.getTypeID() == BlackboardArtifact.ARTIFACT_TYPE.TSK_KEYWORD_HIT.getTypeID())) {
+            updateSheetForAnalysisResult((AnalysisResult) artifact, sheetSet);
```

Comment:
The artifact is used in tupdateSheetForAnalysisResult.  And updateSheetForAnalyResult calls methods that are specific to AnalysisResult.  I don't see away to avoid the cast.

Reply:
I missed the references to the analysis result in the method initially.  My mistake.  Could something like `artifact instanceof AnalysisResult` replace or be added to line 572 to ensure the safety?

Reply:
@rcordovano It seems like what Greg is suggesting is working around a bug in some other part of the code.  What is the point of the artifactType Category if an artifact of type ANALYSIS_RESULT is not in an AnalysisResult object?

Reply:
I believe we typically perform some sort of `instanceof` check before downcasting in the code base.  In this instance, performing an `instanceof` check seems safer and more direct (i.e. we know we can safely cast to `AnalysisResult` because the item is an `AnalysisResult`).  Is there a reason why checking the class type of the `BlackboardArtifact` using the category is preferable in this situation?

Reply:
@rcordovano  & @gdicristofaro I see now that we still have many places that are creating BlackboardArtifact and passing them to the BlackboardArtifactNode constructor (CVT) instead of DataArtifacts and AnalysisResult.

Reply:
@kellykelly3 and @gdicristofaro, places that create BlackboardArtifact objects instead of either DataArtifact or AnalysisResult objects are a problem. I am looking into this in a different context. I'll get back to you on this.